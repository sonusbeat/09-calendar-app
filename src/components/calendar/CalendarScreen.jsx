import { Calendar, momentLocalizer } from 'react-big-calendar'
import moment from "moment";
import "react-big-calendar/lib/css/react-big-calendar.css";

import Navbar from "../ui/Navbar";

const localizer = momentLocalizer(moment)

const events = [{
  title: "Meeting with UI",
  start: moment().toDate(), // new Date()
  end: moment().add( 2, "hours" ).toDate(),
}];

const CalendarScreen = () => {
  return (
    <div className="calendar-screen">
      <Navbar />

      <Calendar
        localizer={ localizer }
        events={ events }
        startAccessor="start"
        endAccessor="end"
      />

    </div>
  );
}

export default CalendarScreen;