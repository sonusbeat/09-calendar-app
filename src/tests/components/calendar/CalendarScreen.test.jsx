import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import CalendarScreen from "../../../components/calendar/CalendarScreen";

// jest.mock("../../../actions/eventAction", () => ({
//     eventStartDelete: jest.fn(),
// }));

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

const initialState = {
    calendar: {
        events: []
    },
    auth: {
        uid: "0123456789",
        name: "John Doe"
    },
    ui: {
        modalOpen: false
    }
};

const store = mockStore( initialState );

// Simular el dispatch para las pruebas
// store.dispatch = jest.fn();

// Se monta el componente respectivamente
const wrapper = mount(
    <Provider store={ store }>
        <CalendarScreen />
    </Provider>
);

describe("Pruebas en <CalendarScreen />", () => {

    test("Deberia de coincidir con el Shapshot", () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

});
