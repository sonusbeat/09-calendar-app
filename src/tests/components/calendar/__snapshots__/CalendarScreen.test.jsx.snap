// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CalendarScreen /> Deberia de coincidir con el Shapshot 1`] = `
<div
  className="calendar-screen"
>
  <div
    className="navbar navbar-dark bg-dark mb-4"
  >
    <span
      className="navbar-brand"
    >
      John Doe
    </span>
    <button
      className="btn btn-outline-danger"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-sign-out-alt fa-w-16 "
        data-icon="sign-out-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       Salir
    </button>
  </div>
  <div
    className="rbc-calendar"
  >
    <div
      className="rbc-toolbar"
    >
      <span
        className="rbc-btn-group"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Hoy
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          &lt;
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          &gt;
        </button>
      </span>
      <span
        className="rbc-toolbar-label"
      >
        junio 2021
      </span>
      <span
        className="rbc-btn-group"
      >
        <button
          className="rbc-active"
          key="month"
          onClick={[Function]}
          type="button"
        >
          Mes
        </button>
        <button
          className=""
          key="week"
          onClick={[Function]}
          type="button"
        >
          Semana
        </button>
        <button
          className=""
          key="day"
          onClick={[Function]}
          type="button"
        >
          Día
        </button>
        <button
          className=""
          key="agenda"
          onClick={[Function]}
          type="button"
        >
          Agenda
        </button>
      </span>
    </div>
    <div
      aria-label="Month View"
      className="rbc-month-view"
      role="table"
    >
      <div
        className="rbc-row rbc-month-header"
        role="row"
      >
        <div
          className="rbc-header"
          key="header_0"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            lun.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_1"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mar.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_2"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mié.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_3"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            jue.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_4"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            vie.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_5"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            sáb.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_6"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            dom.
          </span>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-today"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                31
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                01
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                02
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                03
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                04
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-now rbc-current"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                05
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                06
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                07
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                08
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                09
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                10
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                11
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                12
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                13
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                14
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                15
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                16
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                17
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                18
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                19
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                20
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                21
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                22
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                23
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                24
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                25
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                26
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                27
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                28
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                29
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                30
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                01
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                02
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                03
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                04
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <Portal
    containerInfo={
      <div
        class="ReactModalPortal"
      />
    }
  />
  <button
    className="btn btn-primary fab"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plus fa-w-14 "
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
</div>
`;
